### DISCUSSION TITLE
Entity object Compile Error(ManyToOne, OneToMany..)

### DISCUSSION THREAD

**Discussion by autumn-sun91** [mikro-orm/mikro-orm#5991] [General]
### Describe the bug

I am using IntelliJ. I am trying to establish an association after creating an Entity within the test code written in the mikro-orm github repository, but a compilation error occurs, but no build or runtime errors actually occur. Can you tell me why this is? Likewise for populate, 'Type 'string' is not assignable to type 'never'.' A compilation error occurs, but it does not occur at runtime.

<img width="1186" alt="image" src="https://github.com/user-attachments/assets/60434162-430b-4800-ae96-d704b55db245">

### Reproduction

https://github.com/mikro-orm/mikro-orm/blob/master/tests/entities-sql/Car2.ts

### What driver are you using?

None

### MikroORM version

6.3.7

### Node.js version

v20.17.13

### Operating system

mac

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
I can't help you with screenshots. The code you see in this repository works just fine type-wise, otherwise, the CI wouldn't be green.
